zone/titles.cpp

```diff
@@ -34,349 +35,330 @@ TitleManager::TitleManager() {
 
 bool TitleManager::LoadTitles()
 {
-	Titles.clear();
+	titles.clear();
 
 	std::string query = "SELECT `id`, `skill_id`, `min_skill_value`, `max_skill_value`, "
-                        "`min_aa_points`, `max_aa_points`, `class`, `gender`, `char_id`, "
-                        "`status`, `item_id`, `prefix`, `suffix`, `title_set` FROM titles";
-    auto results = database.QueryDatabase(query);
-	if (!results.Success()) {
+						"`min_aa_points`, `max_aa_points`, `class`, `gender`, `char_id`, "
+						"`status`, `item_id`, `prefix`, `suffix`, `title_set` FROM titles";
+	auto results = database.QueryDatabase(query);
+	if (!results.Success() || !results.RowCount()) {
 		return false;
 	}
 
-	for (auto row = results.begin(); row != results.end(); ++row) {
-        TitleEntry Title;
-		Title.TitleID = atoi(row[0]);
-		Title.SkillID = (EQ::skills::SkillType) atoi(row[1]);
-		Title.MinSkillValue = atoi(row[2]);
-		Title.MaxSkillValue = atoi(row[3]);
-		Title.MinAAPoints = atoi(row[4]);
-		Title.MaxAAPoints = atoi(row[5]);
-		Title.Class = atoi(row[6]);
-		Title.Gender = atoi(row[7]);
-		Title.CharID = atoi(row[8]);
-		Title.Status = atoi(row[9]);
-		Title.ItemID = atoi(row[10]);
-		Title.Prefix = row[11];
-		Title.Suffix = row[12];
-		Title.TitleSet = atoi(row[13]);
-		Titles.push_back(Title);
+	for (auto row : results) {
+		TitleEntry title;
+		title.title_id = std::stoi(row[0]);
+		title.skill_id = (EQ::skills::SkillType) std::stoi(row[1]);
+		title.min_skill_value = std::stoi(row[2]);
+		title.max_skill_value = std::stoi(row[3]);
+		title.min_aa_points = std::stoi(row[4]);
+		title.max_aa_points = std::stoi(row[5]);
+		title.class_id = std::stoi(row[6]);
+		title.gender_id = std::stoi(row[7]);
+		title.character_id = std::stoi(row[8]);
+		title.status = std::stoi(row[9]);
+		title.item_id = std::stoi(row[10]);
+		title.prefix = row[11];
+		title.suffix = row[12];
+		title.titleset = std::stoi(row[13]);
+		titles.push_back(title);
 	}
 
 	return true;
 }
 
-EQApplicationPacket *TitleManager::MakeTitlesPacket(Client *c)
+EQApplicationPacket *TitleManager::MakeTitlesPacket(Client *client)
```

Comment:
We could rewrite this to use [SerializeBuffer](https://github.com/EQEmu/Server/blob/ffa968f64fca86f106bf502e1be2444749a1b675/common/serialize_buffer.h#L9) instead of having to calculate the size.